### PR TITLE
Clean up concurrent

### PR DESCRIPTION
**Pull request by vdemedes**
Benchmark:

screen shot 2016-02-11 at 2 43 12 pm

### CODE REVIEW

File: lib/concurrent.js [hunk 1]
```diff
@@ -10,86 +12,85 @@ function Concurrent(tests, bail) {
 		throw new TypeError('Class constructor Concurrent cannot be invoked without \'new\'');
 	}
 
+	if (!tests) {
+		throw new Error('Concurrent items can\'t be undefined');
```
Comment (sindresorhus) [inline lib/concurrent.js]:
```js
if (!Array.isArray(tests)) {
  throw new TypeError('Expected an array of tests');
}
```

Comment (vdemedes) [inline lib/concurrent.js]: Right, this is better.

File: lib/concurrent.js [hunk 2]
```diff
@@ -10,86 +12,85 @@ function Concurrent(tests, bail) {
...
+	if (result.passed === false) {
+		this.passed = false;
+
+		// only set reason once
+		if (!this.reason) {
+			this.reason = result.reason;
+		}
+
+		if (this.bail) {
+			throw new AvaError('Error in Concurrent while in bail mode');
```
Comment (sindresorhus) [inline lib/concurrent.js]: This error message is unclear. What is the purpose of it?

Comment (vdemedes) [inline lib/concurrent.js]: This error will never be shown to a user, it's just for us. Did not want to leave AvaError without a message. It's for us, like a code comment, to indicate that this error causes execution to bail.

Comment (sindresorhus) [inline lib/concurrent.js]: Oh ok. That makes sense.

**Pull request merged**